### DISCUSSION TITLE
Can I use Django template variables {{ }} in component attributes to fill them dynamically?

### DISCUSSION THREAD

**Discussion by TwistMeister** [django-components/django-components#681] [Q&A] [answered]
Hi, I tried to find an example in the docs, but couldn't find it. So my situation is that I would like to pass on an `ID` to my component, because I'm implementing them in a `for` loop and this helps me identify which instance is which.

This is an example of what I'd like to do, but this results in `{{ item.id }}` being rendered literally instead of the items `ID`.
`{% component "modal" modal_id="{{ item.id }}" size="dialog-medium" %}`

Rendered HTML: `<dialog class="modal dialog-medium" aria-modal="true" role="dialog" id="{{ modal_id }}">`

Any help on how to achieve the desired behaviour is appreciated!

**Comment by EmilStenstrom** [comment]
Hi @TwistMeister,

We need more details to troubleshoot this. What template does your model component use? How does `modal_id="{{ item.id }}"` get transformed into `id="{{ modal_id }}"`?

**Reply by TwistMeister** [reply, marked as answer]
Hi, I eventually got it to work by wrapping it in `{% with %}`-tags and within them parsing and storing the `item.id` in a template variable. Which I then used as a value for the `modal_id` component attribute.

**Reply by EmilStenstrom** [reply]
It would still be interesting to see the actual code here, to understand if there's a bug in our code somewhere. If you could post it, it would be helpful.
